src/SegmentedControl/ToggleButton.tsx

```diff
@@ -22,7 +22,7 @@ export const ToggleButton = <Value,>({
   return (
     <Wrapper
       as="label"
-      data-testId={`value-toggle-${value}`}
+      data-testId={`value-toggle-${value as string}`}
```

Comment:
Would it work if we coerce it to a string instead? underneath this template literal, it should call something similar to `toString` or String.

```suggestion
      data-testId={`value-toggle-${String(value)}`}
```